fix: some scripts

elsongabriel opened the pull request:
# Description

Continues fixing of #1741

- Created a function to check time in levers and changed with `creature:getStorageValue(config.storage) > os.time()`;
- Adjusted save_interval messages;
- Added check to show msg with payment information when traveling as free;
- Fixed loot to custodian;
- Fixed some typos;
- Adjusted reward talkaction, to check cap;
- Fixed adventurer's stone usage;
- Added messages when starting or stopping using exercise weapons;
- Fixed some events of bosses;

Please delete options that are not relevant.
  - [x] Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested

**Test Configuration**:

  - Server Version: Canary 3.1.1
  - Client: Tibia 13.21
  - Operating System: Windows/Ubuntu

## Checklist

  - [x] My code follows the style guidelines of this project
  - [x] I have performed a self-review of my own code
  - [x] I checked the PR checks reports
  - [x] I have commented my code, particularly in hard-to-understand areas
  - [x] I have made corresponding changes to the documentation
  - [x] My changes generate no new warnings
  - [x] I have added tests that prove my fix is effective or that my feature works

elsongabriel committed:
fixes in levers and other fixes.

luanluciano93 commented on data-otservbr-global/scripts/actions/quests/dangerous_depth/gnomish_pesticide.lua:
```diff
@@ -4,7 +4,7 @@ function dangerousDepthPesticide.onUse(player, item, fromPosition, target, toPos
 	end
 
-	if not target or not target:isItem() then
+	if (target == nil) or not target:isItem() then
```
qual a vantagem disso?

elsongabriel replied:
target em alguns casos vem nulo, mesmo com a validação `not target`, e daí estoura erro.  "target nil value"

luanluciano93 commented on data/events/scripts/player.lua:
```diff
@@ -255,7 +255,7 @@ function Player:onMoveItem(item, count, fromPosition, toPosition, fromCylinder,
 		exhaust[playerId] = true
 		addEvent(function()
 			exhaust[playerId] = false
-		end, 2000, playerId)
```
ele pega o playerId externo dentro da função?

elsongabriel replied:
addEvent, não precisa de um terceiro parametro.

luan replied:
> ele pega o playerId externo dentro da função?

Sim, ele pega, isso não tem muito problema. Apesar de ser mais "certinho" passar o parametro e receber dentro ali `function(playerId)`

luanluciano93 commented on data/scripts/reward_chest/boss_death.lua:
```diff
@@ -1,9 +1,10 @@
-	if not corpse or corpse == 0 then
-		return true
+	if not corpse then
+		return false
```
se retornar falso a criatura nao vai morrer, melhor colocar apenas uma mensagem de erro

elsongabriel replied:
a validação não é essa. a criatura já foi morta, mas n possui corpse, então não existe "bau".

moviebr replied:
O return false não vai deixar a criatura morrer.

elsongabriel replied:
corrigido. mantive o return true,

luanluciano93 commented on data/scripts/reward_chest/boss_death.lua:
```diff
@@ -14,14 +15,13 @@ function bossDeath.onDeath(creature, corpse, killer, mostDamageKiller, lastHitUn
-		else
-			corpse:registerReward()
 		end
+		corpse:registerReward()
```
essa função tem como objeto o container, ele precisa ficar dentro da verificação se é ou nao um container

elsongabriel replied:
se entra nesse script boss_death, é porque a creature morta, possui a flag isRewardBoss.

luanluciano93 commented on data/scripts/talkactions/player/bank.lua:
```diff
@@ -71,13 +71,17 @@ local transfer = TalkAction("!transfer")
 
+	local name = split[1]
+	if not name then
+		return true
```
falta uma mensagem de erro

luanluciano93 commented on data/scripts/talkactions/player/reward.lua:
```diff
@@ -24,9 +24,10 @@ local function sendExerciseRewardModal(player)
 
 				local inbox = player:getSlotItem(CONST_SLOT_STORE_INBOX)
-				if inbox and inbox:getEmptySlots() > 0 then
+				if inbox and inbox:getEmptySlots() > 0 and player:getFreeCapacity() >= it.cap then
```
iType:getWeight()

luanluciano93 commented on data/scripts/reward_chest/boss_death.lua:
```diff
@@ -14,14 +15,13 @@ function bossDeath.onDeath(creature, corpse, killer, mostDamageKiller, lastHitUn
 	if monsterType and monsterType:isRewardBoss() then
 		if not corpse.isContainer or not corpse:isContainer() then
-			if corpse.getId then
+			if corpse:getId() ~= nil then
```
qual a vantagem disso?

elsongabriel replied:
por alguma forma, o corpse.getId estava vindo nil em alguns casos, e não conseguia validar e estourava erro.
(mas já está `if corpse.getId then` de novo)

luanluciano93 replied:
Pode deixar os dois pontos, a questão que perguntei é sobre o 'se condição diferente de nil então', melhor usar 'se condição entao'

elsongabriel committed:
bosses: fixes.

elsongabriel committed:
bank: fixes.

elsongabriel committed:
reward: fixes.

elsongabriel committed:
fixes.

moviebr commented on data/libs/functions/player.lua:
```diff
@@ -677,5 +681,8 @@ end
+	if cooldown > os.time() then
+		return false
+	end
+	return true
```
Não é necessário essa verificação aqui, o operador ternário na linha 680 já faz isso

elsongabriel replied:
a forma do ternário, não estava funcionando, aí mantive a forma padrão.

luan commented on data/scripts/reward_chest/boss_death.lua:
```diff
@@ -14,14 +15,13 @@ function bossDeath.onDeath(creature, corpse, killer, mostDamageKiller, lastHitUn
 				logger.warn("[bossDeath.onDeath] Corpse (id: {}, name: {}) for reward boss {} is not a container.", corpse:getId(), corpse:getName(), creature:getName())
 			else
 				logger.warn("[bossDeath.onDeath] Error to get corpseId from boss: {}", creature:getName())
 			end
```
There should probably be a `return` here since we can't do rewards when the boss has no corpse.

elsongabriel replied:
I've been tested with return and had some problems. I'll check again.

luan commented on data-otservbr-global/scripts/actions/quests/forgotten_knowledge/last_lore_lever.lua:
all of these boss levers are being redone in https://github.com/opentibiabr/canary/pull/1783 using the proper lever systems

elsongabriel replied:
When the pr 1783 is available to merge, i'll continue with this pr.

luan commented on data-otservbr-global/scripts/globalevents/customs/save_interval.lua:
```diff
@@ -4,16 +4,10 @@ local function serverSave(interval)
 	saveServer()
-	local message = "Server save complete. Next save in %d %ss!"
-	local messageSingle = "Server save complete. Next save in %d %s!"
+	local message = string.format(SAVE_INTERVAL_CONFIG_TIME > 1 and "Server save complete. Next save in %d %ss!" or "Server save complete. Next save in %d %s!", SAVE_INTERVAL_CONFIG_TIME, SAVE_INTERVAL_TYPE)
+	Game.broadcastMessage(message, MESSAGE_GAME_HIGHLIGHT)
+	logger.info(message)
 	Webhook.sendMessage("Server save", message, WEBHOOK_COLOR_WARNING)
```
The period save is so fast these days we should probably not even broadcast these anymore, it's just spam now.

luan commented on data/items/items.xml:
```diff
@@ -51930,11 +51930,11 @@
+		<attribute key="life leech" value="3"/>
+		<attribute key="mana leech" value="3"/>
+		<attribute key="critical hit" value="3"/>
+		<attribute key="skillboost sword" value="3"/>
```
This indentation was correct before

luan commented on data/libs/functions/player.lua:
```diff
@@ -677,5 +681,8 @@ end
+	if cooldown > os.time() then
+		return false
+	end
+	return true
```
```suggestion
	return cooldown <= os.time()
```

luan reviewed (changes requested):
Please remove changes that conflict with https://github.com/opentibiabr/canary/pull/1783. Also left some other minor comments.

elsongabriel committed:
save_interval.lua: commented broadcast message function.

elsongabriel committed:
items.xml: removed wrong indentation.

elsongabriel committed:
fixed addEvent parameters.

elsongabriel committed:
exercise_training: created message when player starts training or already training.

elsongabriel committed:
adventurers_stone: fixed validation to enable the usage of adventurer's stone (was being used in houses).

elsongabriel committed:
quests e bosses: fixed events name and fixed variables. removed 'rewardBoss' tag from bosses without reward and corpse.

elsongabriel committed:
logCommand improvement.

github-actions[bot] committed:
Lua code format - (Stylua)

elsongabriel committed:
adventurers_stone: fixed validations.

elsongabriel committed:
functions: removed unused function.

sebbesiren committed:
feat: custom pvp system (#1403)

Created two different systems to modify damage values in PvP. 

The first system is damage rate based on levelse. These are controller by three attributes in the config.lua. These have been added to the config.lua.dist.

The second system is damage rates based on vocations. This system is modified using vocations.xml. Added a new tag to the xml named pvp. Can control damage take and dealt per vocation.

Leshyah committed:
fix: the outlaw camp quest (#1809)

luan committed:
fix: charge house rent (#1814)

Closes #987

luan committed:
fix: bestiary kills and boss boost (#1813)

Fixes #1804

Schiffers committed:
feat: random monsters spawn (#1802)

By using this commit you can choose if you want a "dinamic" respawn since it checks the monsters at the same spawn.

Commit:
fix: table.remove function does not work with random indexes (#1801)

dudantas commented on data/events/scripts/player.lua:
```diff
@@ -253,9 +253,9 @@ function Player:onMoveItem(item, count, fromPosition, toPosition, fromCylinder,
 		end
 		exhaust[playerId] = true
-		addEvent(function()
+		addEvent(function(playerId)
```
If you removed it below, you could also remove it here.
```suggestion
		addEvent(function()
```

elsongabriel replied:
> If you removed it below, you could also remove it here.

I thought I had presented Luan's suggestion, I'll do it now.

luanluciano93 replied:
```suggestion
		addEvent(function(playerIdInEvent)
			exhaust[playerIdInEvent] = false
		end, 2000, playerId)
```

elsongabriel committed:
ssa exhaust fixes.

dudantas committed:
Merge branch 'main' into hotfix/fix-scripts

luan commented on data-otservbr-global/scripts/globalevents/customs/save_interval.lua:
```diff
@@ -4,16 +4,10 @@ local function serverSave(interval)
-	local message = "Server save complete. Next save in %d %ss!"
-	local messageSingle = "Server save complete. Next save in %d %s!"
+	local message = string.format(SAVE_INTERVAL_CONFIG_TIME > 1 and "Server save complete. Next save in %d %ss!" or "Server save complete. Next save in %d %s!", SAVE_INTERVAL_CONFIG_TIME, SAVE_INTERVAL_TYPE)
+	--Game.broadcastMessage(message, MESSAGE_GAME_HIGHLIGHT)
```
uncomment 
```suggestion
	Game.broadcastMessage(message, MESSAGE_GAME_HIGHLIGHT)
```

luan commented on data/libs/exercise_training.lua:
```diff
@@ -55,6 +55,7 @@ function ExerciseEvent(playerId, tilePosition, weaponId, dummyId)
 	end
 
 	if player:isTraining() == 0 then
+		player:sendTextMessage(MESSAGE_FAILURE, "You left training!")
```
You've stopped training.
```suggestion
		player:sendTextMessage(MESSAGE_FAILURE, "You've stopped training.")
```

luan committed:
Update data-otservbr-global/scripts/globalevents/customs/save_interval.lua
